### PR TITLE
Add packaging guide #5182

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Initial version of a packaging guide

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -56,6 +58,8 @@ Nim from source using ``gcc``, ``git`` and the ``koch`` build tool.
 For most users, installing the latest stable version is enough. Check out
 the installation instructions on the website to do so: https://nim-lang.org/install.html.
 
+For package mantiners: see [packaging guidelines](https://nim-lang.github.io/Nim/packaging.html).
```
Comment (dom96) [inline readme.md]:
```suggestion
For package maintainers: see [packaging guidelines](https://nim-lang.github.io/Nim/packaging.html).
```

File: doc/packaging.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+   ./koch boot -d:release
+
+   # optionally generate docs
+   ./koch docs
+   ./koch tools -d:release
+
+   # extract files to be really installed
+   ./install.sh <tempdir>
+
+The "compiler" directory contains compiler sources and should not be part of the compiler binary package.
```
Comment (dom96) [inline doc/packaging.rst]: Are we certain of this?

Comment (FedericoCeratto) [reply doc/packaging.rst]: See https://github.com/nim-lang/Nim/issues/4866#issuecomment-271194805 and related issue.

**Comment by genotrance** [comment]
Per https://github.com/nim-lang/Nim/issues/5912, install.sh isn't supported but its still present. What's the direction here?

I'm working on getting Linux binary builds enabled with [Holy Build Box](https://github.com/phusion/holy-build-box) and it is coming along nicely. However, a release build should contain the full source so I won't be using install.sh. We need to update `koch xz` to include the Linux binaries in the generated xz file.

**Comment by Araq** [comment]
install.sh is for package builders and supposed to be created by ``koch distrohelper``.

**Comment by Araq** [comment]
Merging now, we can always improve it later.

File: doc/packaging.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+   # optionally generate docs into doc/html
+   ./koch docs
+
+   ./koch tools -d:release
+
+   # extract files to be really installed
+   ./install.sh <tempdir>
+
+   # also include the tools
+   for fn in nimble nimsuggest nimgrep; do cp ./bin/$fn <tempdir>/nim/bin/; done
```
[the remark this answers is not in the document]
Comment (FedericoCeratto) [reply doc/packaging.rst]: Will do. (you meant nimgrep?)

File: doc/packaging.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+What to install:
+
+- The expected stdlib location is /usr/lib/nim
+- Global configuration files under /etc/nim
+- Optionally: manpages, documentation, shell completion
```
[the remark this answers is not in the document]
Comment (FedericoCeratto) [reply doc/packaging.rst]: The completions are contributed. I'll update.

File: doc/packaging.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (alaviss) [reply doc/packaging.rst]: `nimfind` is a new tool, it's in `tools/nimfind.nim`

[the remark this answers is not in the document]
Comment (FedericoCeratto) [reply doc/packaging.rst]: For the record: nimfind and nim-gdb are not shipped with in any 0.19.* and will be in 0.20

[the remark this answers is not in the document]
Comment (alaviss) [reply doc/packaging.rst]: Well, but this is a part of the bleeding edge docs, is't it?

**Pull request merged**